### ISSUE TITLE
Restarts not  working for live or die spectra

### ISSUE THREAD

**Issue by kslong** [sirocco-rt/sirocco#833]
For the following .pf file, if you edit disk.pf so that the output spectral cycles ate say 300 and run the model too completion, and then edit the output cycles to 1000, you will see the spectra are not properly normalized.  

I am not sure whether this is a live or die problem or a macro atom problem.

[z.fail.210322.tar.gz](https://github.com/agnwinds/python/files/6185398/z.fail.210322.tar.gz)

Sigh ....

**Comment by kslong** [comment]
The problem goes away if you one is in extract and not live or die mode

**Comment by kslong** [comment]
Here is what happens when you run some simple cv models, all in live or die mode.  The 3 models are

- one run for 20 spectral cycles
- one run for 10 spectral cycles
- one starting with the previous run, which has an additional 10 cycles

![compare](https://user-images.githubusercontent.com/2530901/117025623-afea5380-acc0-11eb-8b15-99c8f3d7e57a.png)

The normalization of the restarted model is completely wrong.  

Here is a set of files, which should reproduce these (incorrect) results.

[z.xxx.210504.tar.gz](https://github.com/agnwinds/python/files/6422064/z.xxx.210504.tar.gz)

**Comment by kslong** [comment]
The normalization problem does not depend on the number of threads, but as one might expect it does depend on how many spectral cycles one runs.  For the example above, the "correct" average value in the spectrum is 1.9e-11.  For 10 extra cycles on the restart, one gets 1.9e-9.  For 1 extra cycle on the restart, one gets 3.6e-10, whereas for 20 extra one gets 2.5e-9.

**Comment by kslong** [comment]
This problem was obscure.  It turned out to be that on restarts, select_extract was being initialized to extract mode rather than live or die.  On restarts, the code does not re-read most of the keywords in the .pf file and setting the mode to extract rather than live or die caused a major problem since the normalization for the spectra was set for what it should have been in live or die mode.  The two lines in python.c, where the initialization took place were actually unnecessary and removing them allowed the program to work correctly.  This error had been around since at least 2015; the reason we did not find it was probably due to this particular mode of operation, namely restarting with live or die, had not been encountered.  Here is the new version of the figure that is above

![compare](https://user-images.githubusercontent.com/2530901/121237338-9064be80-c85c-11eb-9124-ff1ce9f6cb9b.png)

The corrections have been merged into dev
